### ISSUE TITLE
Linter erroneously signaling an error on working code

### ISSUE THREAD

**Issue by ptournet** [julia-vscode/julia-vscode#2575]
With vscode 1.62.2 and julia-vscode 1.5.5,  the following code, we can see that the linter is signaling an error on the definition of the function add :

![image](https://user-images.githubusercontent.com/4687542/142486999-b3bba4fa-ba16-4497-a0d9-5c2379aafa13.png)

The error is the following :

![image](https://user-images.githubusercontent.com/4687542/142487273-fef59ed3-4f8e-4282-a9ea-eb051e03fdee.png)

Yet the code is working as expected :

![image](https://user-images.githubusercontent.com/4687542/142487407-7f7b8c8f-4c7c-406c-9a46-f654aa0f27e3.png)

**Comment by pfitzseb** [comment]
I have never seen this pattern of manually subtyping `Function` before, crazy :D

Just out of curiousity: Why are you doing this? For being able to dispatch on `ArithmeticFunction`?

**Comment by ptournet** [comment]
Yes that's exactly the purpose : being able to dispatch on subtypes of Function ! I was hoping to find a standard way of doing that, but that one worked and it's not that too bad... ;-)

**Comment by pfitzseb** [comment]
I think the typical pattern for that is dispatching on `typeof(add)` (as in `foo(::typeof(add), args...)`).

**Comment by davidanthoff** [comment]
The language server backend was rewritten in v1.220.0: StaticLint.jl and SymbolServer.jl were folded into JuliaWorkspaces, and completion, hover, signature help, go-to-definition and find-references were reimplemented. This report predates that rewrite, so the code that produced it no longer exists and the behaviour cannot be confirmed against the current engine.

Closing as outdated. If you still see this on a current release, please open a fresh issue with a reproducer and we will look at it against the new engine.
